mixer/pkg/attribute/bag.go

```diff
@@ -32,3 +52,42 @@ type Bag interface {
 	// Done indicates the bag can be reclaimed.
 	Done()
 }
+
+// Equal compares two attribute values.
+func Equal(this, that interface{}) bool {
+	switch x := this.(type) {
+	case int64, int, string, float64, bool:
+		return x == that
+	case time.Time:
+		if y, ok := that.(time.Time); ok {
+			return x.Equal(y)
+		}
+	case time.Duration:
+		if y, ok := that.(time.Duration); ok {
+			return x == y
+		}
+	case []byte:
+		if y, ok := that.([]byte); ok {
+			return reflect.DeepEqual(x, y)
+		}
+	case StringMap:
+		if y, ok := that.(StringMap); ok {
+			return x.Equal(y)
```

Comment:
Shouldn't this use relfect.DeepEquals like compareAttributes was?

Reply:
It is DeepEquals under the hood. The intent is to compare as-value, ignoring the parent proto bag.

Reply:
The new function is equivalent to the old one and made public (for use beyond test utils)